### PR TITLE
Enable updating of channelsData dynamically

### LINKED ISSUE #10: It's not possible to change channels data bindings dynamically in app life cycle
In service `pusher-base.js` we define `channelsData` that stores information to which channels and events we should subscribe to Pusher. However, it's not possible to change that dynamically in application life cycle - e.g. between route transitions.

Change of `channelsData` should trigger subscription to new events and channels and unsubscription from channels and events that were deleted from this object.

### PR DESCRIPTION
**Pull request by jbandura**
This PR adds the functionality of dynamically updating `channelsData` property. It exposes three new functions:
- `updateChannelsData` - replaces existing channelsData with the one provided
- `addChannelsData` - adds single or (if argument is array) multiple channel data
- `removeChannel` - removes specified channel

Closes #10

### CODE REVIEW

**Comment by jniechcial** [comment]
Great job @jbandura ! @Pasikonik What do you think?

File: addon/services/pusher-base.js [hunk 1]
```diff
@@ -66,24 +90,36 @@ export default Service.extend(Ember.Evented, Checker, {
...
 
-  _setSubscriptionsEndEvents() {
-    this.get('channelsData').forEach((singleChannel) => {
+  _manageChannelsChange(oldChannelsData, newChannelsData) {
+    const {
+      channelsToSubscribe,
+      channelsToUnsubscribe
+    } = channelsDiff(oldChannelsData, newChannelsData);
+    this._subscribeChannels(channelsToSubscribe);
+    this._unsubscribeChannels(channelsToUnsubscribe);
```
Comment (Pasikonik) [inline addon/services/pusher-base.js]: try to no resubscribe all channel in every change in channelsData

Comment (jbandura) [inline addon/services/pusher-base.js]: @Pasikonik actually the `channelsDiff` utility function selects only channels that are not yet subscribed: https://github.com/netguru/ember-pusher-guru/pull/13/files#diff-5043340fe86d0025314bb5ec04f3c81eR14

Comment (Pasikonik) [inline addon/services/pusher-base.js]: 👍

**Pull request merged**